server/src/main/kotlin/suwayomi/tachidesk/graphql/mutations/IReaderSourceMutation.kt

```diff
@@ -0,0 +1,186 @@
+                    novels = novelsPage.mangas.map { IReaderNovelType(it, sourceId) },
+                    hasNextPage = novelsPage.hasNextPage,
+                )
+            }
+        }
+    }
+
+    data class FetchIReaderChaptersInput(
+        val clientMutationId: String? = null,
+        @GraphQLDescription("Source ID (required if using novelUrl)")
+        val source: Long? = null,
+        @GraphQLDescription("Novel URL/key from the source (provide this OR novelId)")
+        val novelUrl: String? = null,
+        @GraphQLDescription("Database novel ID (provide this OR novelUrl + source)")
+        val novelId: Int? = null,
+    )
+
+    data class FetchIReaderChaptersPayload(
+        val clientMutationId: String?,
+        val chapters: List<IReaderChapterType>,
+    )
+
+    @RequireAuth
+    @GraphQLDescription("Fetch chapters for a novel from an IReader source")
+    fun fetchIReaderChapters(input: FetchIReaderChaptersInput): CompletableFuture<DataFetcherResult<FetchIReaderChaptersPayload?>> {
+        val (clientMutationId, source, novelUrl, novelId) = input
+
+        return future {
+            asDataFetcherResult {
+                val (resolvedSourceId, resolvedNovelUrl) =
+                    when {
+                        novelId != null -> {
+                            // Look up novel from database
+                            val novel =
+                                transaction {
+                                    IReaderNovelTable
+                                        .selectAll()
+                                        .where { IReaderNovelTable.id eq novelId }
+                                        .firstOrNull()
+                                } ?: throw IllegalArgumentException("Novel with ID $novelId not found")
+                            novel[IReaderNovelTable.sourceReference] to novel[IReaderNovelTable.url]
+                        }
+                        novelUrl != null && source != null -> {
+                            require(novelUrl.isNotBlank()) { "Novel URL cannot be empty" }
+                            source to novelUrl
+                        }
+                        else -> throw IllegalArgumentException("Either novelId OR (novelUrl + source) must be provided")
+                    }
+
+                val chapters = IReaderNovel.getChapterList(resolvedSourceId, resolvedNovelUrl)
+
+                FetchIReaderChaptersPayload(
+                    clientMutationId = clientMutationId,
+                    chapters = chapters.map { IReaderChapterType(it) },
+                )
+            }
+        }
+    }
+
+    data class FetchIReaderChapterContentInput(
+        val clientMutationId: String? = null,
+        @GraphQLDescription("Source ID (required if using chapterUrl)")
+        val source: Long? = null,
+        @GraphQLDescription("Chapter URL/key from the source (provide this OR chapterId)")
+        val chapterUrl: String? = null,
```

Comment:
This and the sourceId should be removed, at this point we should be relying only on the chapterId. Use that to get the sourceId.